components/org.wso2.carbon.identity.workflow.engine/src/main/java/org.wso2.carbon.identity.workflow.engine/DefaultWorkflowEventRequestService.java

```diff
@@ -0,0 +1,151 @@
+import org.wso2.carbon.identity.workflow.engine.dao.impl.WorkflowEventRequestDAOImpl;
+import org.wso2.carbon.identity.workflow.engine.exception.WorkflowEngineException;
+import org.wso2.carbon.identity.workflow.engine.exception.WorkflowEngineRuntimeException;
+import org.wso2.carbon.identity.workflow.engine.util.WorkflowEngineConstants;
+import org.wso2.carbon.identity.workflow.mgt.WorkflowExecutorManagerService;
+import org.wso2.carbon.identity.workflow.mgt.WorkflowExecutorManagerServiceImpl;
+import org.wso2.carbon.identity.workflow.mgt.WorkflowManagementService;
+import org.wso2.carbon.identity.workflow.mgt.WorkflowManagementServiceImpl;
+import org.wso2.carbon.identity.workflow.mgt.bean.Parameter;
+import org.wso2.carbon.identity.workflow.mgt.bean.RequestParameter;
+import org.wso2.carbon.identity.workflow.mgt.bean.Workflow;
+import org.wso2.carbon.identity.workflow.mgt.bean.WorkflowAssociation;
+import org.wso2.carbon.identity.workflow.mgt.dto.WorkflowRequest;
+import org.wso2.carbon.identity.workflow.mgt.exception.InternalWorkflowException;
+import org.wso2.carbon.identity.workflow.mgt.exception.WorkflowException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+public class DefaultWorkflowEventRequestService implements DefaultWorkflowEventRequest {
+
+    public void addApproversOfRequests(WorkflowRequest request, List<Parameter> parameterList) {
+
+        WorkflowEventRequestDAO workflowEventRequestDAO = new WorkflowEventRequestDAOImpl();
+        String taskId = UUID.randomUUID().toString();
+        String eventId = getRequestId(request);
+        String workflowId = getWorkflowId(request);
+        String approverType = null;
+        String approverName = null;
+        int currentStepValue = getStateOfRequest(eventId, workflowId);
+        if (currentStepValue == 0) {
+            createStatesOfRequest(eventId, workflowId, currentStepValue);
+        }
+        currentStepValue += 1;
+        updateStateOfRequest(eventId, workflowId);
+        List<WorkflowAssociation> associations = getAssociations(request);
+        for (WorkflowAssociation association : associations) {
+            for (Parameter parameter : parameterList) {
+                if (parameter.getParamName().equals(WorkflowEngineConstants.ParameterName.USER_AND_ROLE_STEP)) {
+                    String[] stepName = parameter.getqName().split("-");
+                    int step = Integer.parseInt(stepName[2]);
+                    if (currentStepValue == step) {
+                        approverType = stepName[stepName.length - 1];
+
+                        String approver = parameter.getParamValue();
+                        String[] approvers = approver.split(",");
+                        if (approvers != null) {
+                            List<String> approverList = Collections.singletonList(approver);
+                            String stepValue = WorkflowEngineConstants.ParameterName.USER_AND_ROLE_STEP + "-step-" +
```

Comment:
Let's use constants for pattern matching like "-step-"

Reply:
Addresses with this commit https://github.com/wso2-incubator/identity-workflow-simple/pull/1/commits/0f8bc6df9d488542344585725d992c952c1b640b